### ISSUE TITLE
Error "Unable to load DLL 'cvextern'" when using EmguCV 3.4.3.3016 on a Windows N version

### ISSUE THREAD

**Issue by bennycoomans** [emgucv/emgucv#187]
My company uses the commercial version of EmguCV. We have used version 2.4.2.1777 for a long time, because it just works and we don't need a lot of fancy features. We are now trying to upgrade to the newest version, 3.4.3.3016. On my development PC, everything works as expected, apart from some minor changes in the API. I then tried deploying our software to a virtual test environment and encountered the error "Unable to load DLL 'cvextern'". After some googling and using Dependency Walker, I discovered that EmguCV, through OpenCV, depends on the Windows Media Feature pack. The virtual Windows environments that I deployed to are indeed Windows 7 and Windows 10 N versions. After installing the Windows Media Feature pack for the respective Windows versions (there are even different versions of the pack per Windows 10 version, 1709, 1803 and 1809), the problem was resolved. However, we are very hesitative about enforcing dependencies to our customers, and at this point I don't even know whether or not Windows Server installations contain this pack by default.

Long story short, my primary question is this: is my conclusion correct that the only way to get EmguCV to work is by installing the Windows Media Feature pack? Or is there some way around this dependency? If there is no way around it, We may have to look for a possibility to get a refund.

A secondary question is of course if this dependency can be mentioned in the documentation somewhere, so any new or updating user can decide if the new dependency is acceptable.

**Comment by emgucv** [comment]
Emgu CV 3.4 release has Microsoft Media Foundation (MSMF) video backend support enabled for the VideoCapture class.  MSMF is pre-installed for Windows 10 Home or Windows 10 Pro releases. For Windows N or certain Windows Server releases, it may require you to download and install Windows Media Feature pack.

We have updated the release note to mention this extra dependency:
http://www.emgu.com/wiki/index.php/Version_History#Dependency 

We can build a custom release for you that do not include MSMF video backend for a fee. If you would like to proceed please send us an email to support@emgu.com and we can continue from there. 

If the solution is not acceptable, please let us know and we could issue a full refund.

**Comment by bennycoomans** [comment]
Thanks for your reply. I've discussed it with my team. A main reason for updating from 2.4.2 was the fact that NuGet packages are now available. I assume that a custom build will not be available through NuGet, and we do not want to pay a fee (and request a custom build whenever we want to upgrade), so the custom build is not really an option.

We then figured that we might as well use an older version from NuGet. However, versions 3.4.1 and 3.3.0 do not seem to be strongly signed, so we cannot use them. Version 3.2.0 does not have that problem, but according to Dependency Walker, that version already depends on MSMF. Is that correct? If so, we are running out of options and I will email you concerning a refund.

The only thing left for us then is the suggestion to make it part of your build process to release an MSMF and a non-MSMF version and release them separately. That would make it possible for us to update to a more recent version in the future.

**Comment by emgucv** [comment]
I can create a NuGet Pacakge for the custom build such that you can put it in a network shared folder, and your team will be install it through nuget package manager.

The nuget package is not signed but the binary that comes with the nuget package is strongly signed. 

Version 3.2.0 already depends on MSMF. It is added by popular demand. Non-MSMF release will not be include in the standard release in the short term unless there are stronger demand for it. Making non-MSMF release on top of MSMF release will double the build and release effort. Non-MSMF release will be a custom build at this moment. 

Sorry for not being able to accommodate your request, Please send the order number to support@emgu.com for a refund.

**Comment by bennycoomans** [comment]
Thank you for your answers and ideas. I understand your position, so for now we will remain at the old version and I sent an email for a refund.

**Comment by 836257514** [comment]
@emgucv we also want this Non-MSMF release version too！
